addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -186,14 +365,82 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @see this.selectClass for params
      */
-    style: function (previewMode, widgetValue, params) {
-        this._rpc({
-            route: '/shop/change_styles',
-            params: {
-                'id': this.productTemplateID,
-                'style_id': params.possibleValues[params.possibleValues.length - 1],
-            },
+    async setRibbon(previewMode, widgetValue, params) {
+        this.$target[0].dataset.ribbonId = widgetValue;
+        this.trigger_up('set_product_ribbon', {
+            templateId: this.productTemplateID,
+            ribbonId: widgetValue || false,
         });
+        const ribbon = this.ribbons[widgetValue] || {name: '', color: '', text_color: '', html_class: ''};
+        const $ribbons = $(`[data-ribbon-id="${widgetValue}"] .o_ribbon:not(.o_wsale_ribbon_dummy)`);
+        $ribbons.text(ribbon.name);
+        $ribbons.attr('class', 'o_ribbon');
```

Comment:
Would this be that costly to support custom code adding custom classes here? Just remove all html_class of all ribbons?